src/viewport.js

```diff
@@ -124,32 +124,44 @@ export default class Viewport {
    * @param {Array} lngLatZ - [lng, lat] or [lng, lat, Z]
    * @param {Object} opts - options
    * @param {Object} opts.topLeft=true - Whether projected coords are top left
-   * @return {Array} - [x, y] or [x, y, z] in top left coords
+   * @return {Array} - screen coordinates [x, y] or [x, y, z], z as pixel depth
    */
-  project(xyz, {topLeft = false} = {}) {
+  project(xyz, {topLeft = true} = {}) {
     const [x0, y0, z0 = 0] = xyz;
     assert(Number.isFinite(x0) && Number.isFinite(y0) && Number.isFinite(z0), ERR_ARGUMENT);
 
     const [X, Y] = this.projectFlat([x0, y0]);
     const v = transformVector(this.pixelProjectionMatrix, [X, Y, z0, 1]);
 
     const [x, y] = v;
-    const y2 = topLeft ? this.height - y : y;
-    return xyz.length === 2 ? [x, y2] : [x, y2, 0];
+    const y2 = topLeft ? y : this.height - y;
+    return xyz.length === 2 ? [x, y2] : [x, y2, v[2]];
   }
 
   /**
    * Unproject pixel coordinates on screen onto world coordinates,
    * (possibly [lon, lat]) on map.
    * - [x, y] => [lng, lat]
    * - [x, y, z] => [lng, lat, Z]
-   * @param {Array} xyz -
+   * @param {Array} xyz - screen coordinates, z as pixel depth
+   * @param {Object} opts - options
+   * @param {Object} opts.topLeft=true - Whether projected coords are top left
+   * @param {Object} opts.targetZ=0 - If pixel depth is unknown, targetZ is used as
+   *   the elevation plane to unproject onto
    * @return {Array} - [lng, lat, Z] or [X, Y, Z]
    */
-  unproject(xyz, {topLeft = false} = {}) {
-    const [x, y, targetZ = 0] = xyz;
+  unproject(xyz, {topLeft = true, targetZ} = {}) {
```

Comment:
where do we define targetZ's default as 0?

Reply:
When we use it (line#174). The original value of `targetZ` is used to determine whether the return value should be 2-component or 3-component.